lib/vector-core/src/event/metric/data.rs

```diff
@@ -9,6 +9,9 @@ pub struct MetricData {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub timestamp: Option<DateTime<Utc>>,
 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub interval_ms: Option<u64>,
```

Comment:
Some questions about this representation:
1. Would `u32` be sufficient here? That would allow for an interval of up to 49 days.
2. Would it ever be zero? `Option<NonZero*>` would have a more efficient representation.
3. Would it make sense to bundle this with the timestamp into a new struct to reduce the number of parts for `from_parts`/`into_parts`?